src/wallet/mod.rs

```diff
@@ -4425,11 +4426,9 @@ pub(crate) mod test {
             result.is_err(),
             "Signing should have failed because the TX uses non-standard sighashes"
         );
-        assert!(
-            matches!(
-                result.unwrap_err(),
-                Error::Signer(SignerError::NonStandardSighash)
-            ),
+        assert_matches!(
+            result.unwrap_err(),
```

Comment:
You can avoid the `unwrap_err`, and match on `Err(Error::Signer(SignerError::NonStandardSighash))`